cs-mail/src/main/java/io/cloudslang/content/mail/sslconfig/SSLUtils.java

```diff
@@ -82,4 +91,188 @@ public static KeyStore createKeyStore(final URL url, final String password)
+                if (isEmpty(trustPassword)) {
+                    trustPassword = SecurityConstants.DEFAULT_PASSWORD_FOR_STORE;
+                }
+            } else {
+                trustPassword = null;
+            }
+            useTrustCert = storeExists;
+        } else {
+            if (!trustAllRoots) {
+                if (!trustKeystore.startsWith(Constants.HTTP))
+                    trustKeystore = Constants.FILE + trustKeystore;
+                useTrustCert = true;
+            }
+
+        }
+
+        TrustManager[] trustManagers = null;
+        KeyManager[] keyManagers = null;
+
+        if (trustAllRoots) {
+            trustManagers = new TrustManager[]{new EasyX509TrustManager()};
+        }
+
+        if (useTrustCert) {
+            KeyStore trustKeyStore = SSLUtils.createKeyStore(new URL(trustKeystore), trustPassword);
+            trustManagers = SSLUtils.createAuthTrustManagers(trustKeyStore);
+        }
+        if (useClientCert) {
+            KeyStore clientKeyStore = SSLUtils.createKeyStore(new URL(keystore), keystorePassword);
+            keyManagers = SSLUtils.createKeyManagers(clientKeyStore, keystorePassword);
+        }
+
+        SSLContext context = SSLContext.getInstance(SecurityConstants.SSL);
+        context.init(keyManagers, trustManagers, new SecureRandom());
+        SSLContext.setDefault(context);
+    }
+
+
+    static Store connectUsingSSL(Properties props, Authenticator auth, GetMailInput input) throws MessagingException {
+        Store store = configureStoreWithSSL(props, auth, input);
+        store.connect();
+        return store;
+    }
+
+
+    public static Store configureStoreWithSSL(Properties props, Authenticator auth, MailInput input) throws NoSuchProviderException {
+        configureWithSSL(props, input);
+        URLName url = new URLName(input.getProtocol(), input.getHostname(), input.getPort(), StringUtils.EMPTY,
+                input.getUsername(), input.getPassword());
+        Session session = Session.getInstance(props, auth);
+        return session.getStore(url);
+    }
+
+    private static void configureWithSSL(Properties props, MailInput input) {
+        props.setProperty(PropNames.MAIL + input.getProtocol() + PropNames.SOCKET_FACTORY_CLASS, SecurityConstants.SSL_SOCKET_FACTORY);
+        props.setProperty(PropNames.MAIL + input.getProtocol() + PropNames.SOCKET_FACTORY_FALLBACK, String.valueOf(false));
+        props.setProperty(PropNames.MAIL + input.getProtocol() + PropNames.PORT, String.valueOf(input.getPort()));
+        props.setProperty(PropNames.MAIL + input.getProtocol() + PropNames.SOCKET_FACTORY_PORT, String.valueOf(input.getPort()));
+    }
+
+
+    public static Store tryTLSOtherwiseTrySSL(Authenticator auth, Properties props, GetMailInput input) throws MessagingException {
+        Store store = configureStoreWithTLS(props, auth, input);
+        try {
+            store.connect(input.getHostname(), input.getUsername(), input.getPassword());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

This code contains a potential information leakage in the error handling for the following call(s): &#39;MailInput.getPassword()&#39; (line 213). You are handling this error data through: &#39;Store.connect()&#39; (line 213) [with catch clause(s) at: [line 214]]. Please use your organization-specific sensitive information handling loggers or redaction. [Learn more about information leakage](https://cwe.mitre.org/data/definitions/209.html) (external link).